packages/react-storage/src/components/StorageImage/StorageImage.tsx

```diff
@@ -15,23 +15,17 @@ export const StorageImage = ({
   onStorageGetError,
   ...rest
 }: StorageImageProps): JSX.Element => {
-  const options = React.useMemo(
-    () => ({
-      accessLevel,
-      identityId,
-    }),
-    [accessLevel, identityId]
-  );
-
-  const errorConfig = React.useMemo(
+  const useStorageURLParams = React.useMemo(
```

Comment:
Nit: would only memoize the `options` and pass the other values inline to `useStorageURL` to scope the memoization value to the `options` related values

Reply:
Would also need to memoize the `onStorageGetError` with `useCallback`?

Reply:
No since `onStorageGetError` is just passed directly to `useStorageURL` nor is `onStorageGetError` extended by within `StorageImage` (by say,  wrapping within another callback) there is no need for it to be memoized as there is no update to its reference within `StorageImage`. 

Given the scenario of a user defining an `onStorageGetError` callback that has not been memoized before being passed as a prop to `StorageImage` (`onStorageGetError` does not have a stable reference):

- including `onStorageGetError` in the `useStorageURLParams` with `options` would have the unintended consequence of causing `useStorageURLParams` to be re-evaluated on each render

 - wrapping in own `useCallback` would result on re-evaluation on each render

Additionally, since the `useEffect` in `useStorageGetURL` only calls `Storage.get` if `key` is updated the reference of `onStorageGetError` changing will not cause an additional call to `Storage.get`.